### ISSUE TITLE
Replying from notification always uses SIM1

### ISSUE THREAD

**Issue by Cwpute** [FossifyOrg/Messages#154]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Android 11

### Affected device model

Nokia TA-1157

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Have a SIM card inserted in slot number 2 (SIM2) in your phone
2. Set a conversation to send messages using SIM2
3. Receive a message on SIM2 for that conversation
4. From the notification that pops up, tap "Reply" and send a message

### Expected behavior

App should answer using the preferred SIM specified by the user in the conversation with the SMS sender, or at the very least automatically defaults to the SIM that the message was sent to.

### Actual behavior

App will always send the message via SIM1, even if there is no SIM card inserted in the SIM slot number 1.

### Additional information

This is critical for me as some of my clients won't look at messages from phone numbers they don't recognize. So if my app sends replies with the wrong SIM card, clients might never receive critical information i sent them and assume they have received and read as being from me.

originally reported by @Zoyolin in the SMT repository: https://github.com/SimpleMobileTools/Simple-SMS-Messenger/issues/723
> I've noticed that when I reply to a sms received in the notification, the send fails as it attempts to use my first sim card. In the main sms view, sim card 2 is the default one and also system wide for sms (and internet and calls).
I'm on android 11 on a fairphone 3. Build 8901.4.A.0023.0 release-keys.

**Comment by WebworkrNet** [comment]
**Affected Android/Custom ROM version**
Android 14

**Affected device model**
Motorola moto g34 5G (XT2363-3)

See also closed discussion https://github.com/FossifyOrg/Messages/discussions/378.

**Comment by Artiom-M** [comment]
Similar problem on Android 15 / Pixel 7 Pro / Graphene OS

Just in my case default is SIM1 but when replying from notification it uses SIM2

**Comment by Cwpute** [comment]
Interesting. Have you specifically set your SIM1 to be the default ? Have you done anything that would tell your device or app to use a specific SIM card over another ?
I'll take a look myself.

**Comment by Artiom-M** [comment]
Yes, system level default is SIM1. And stock Messaging app works well.

**Comment by NetMan134** [comment]
I have to say, I sometimes fell victim to this bug, I've got 2 SIMs, one for mainly (cheap) internet, one for mainly (cheap) calls and SMS.
The SIM which I mainly use for calls and SMS sits in the SIM2 slot, and the one for internet, in the SIM1.
Now, when I get an SMS message (to SIM2) from a new/unknown number, and reply to it quickly (without paying attention if it goes out of SIM2), it goes out from SIM1...
Shouldn't the reply function work that it checks which SIM got the message, and reply with the same one?

<sup>Aside from that, the apps ecosystem is really great, good job Fossify!</sup>

**Comment by higheraims** [comment]
Seeing the same behavior on Android 16 GrapheneOS. Installed via FDroid.

I'm using eSIM as default. Within the app it correctly replies to texts on the SIM being used for that conversation. From a notification, however, the physical SIM which is SIM1 is used. 

This has had financial consequences as my carrier is charging 70 cents per text for roaming. So my workaround will be to go back to the GrapheneOS stock Messages app or another alternative if necessary.
